### PR TITLE
refactor(managers): move regex in `custom` folder

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Moved regex manager in custom folder and modified the code in some test files, `docs/manager.ts` and manager-fingerprint generation as they depend on the file location of manager files

- Also created the file `custom/index.ts` and exported constants and `extractPackageFile` fn from it to satisfy the manager module validation in tests.

- I haven't included `custom` in the `manager/api.ts`, since`custom` is only a wrapper for custom managers and not an actual manager.

## Context
Part of https://github.com/renovatebot/renovate/issues/19066
Tracked here: https://github.com/renovatebot/renovate/issues/19066#issuecomment-1666409284

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/custom/regex/index.spec.ts [hunk 1]
```diff
@@ -144,7 +144,7 @@ describe('modules/manager/regex/index', () => {
   });
 
   it('extracts and does not apply a registryUrlTemplate if the result is an invalid url', async () => {
-    jest.mock('../../../logger');
+    jest.mock('../../../../logger');
```
Comment (viceice) [inline lib/modules/manager/custom/regex/index.spec.ts:147]: this should be removed, logger is globally mocked.

File: tools/generate-imports.mjs [hunk 2]
```diff
@@ -152,13 +152,18 @@ async function generateHash() {
       .map((file) => file.name);
 
     for (const manager of managers) {
-      const hash = await getManagerHash(manager);
+      const hash = await getManagerHash(
+        manager === 'custom' ? 'custom/regex' : manager
```
Comment (viceice) [inline tools/generate-imports.mjs:156]: 🤔 looks wrong to me. didn't we concluded to name it `custom.regec`?

**Review by viceice — COMMENTED** [review]
did I loose something or is this just a file move refactor PR?

**Comment by RahulGautamSingh** [comment]
> did I loose something or is this just a file move refactor PR?

just a file move refactor

File: tools/generate-imports.mjs (hunk 2 above)
Comment (RahulGautamSingh) [reply tools/generate-imports.mjs:156]: This PR doesn't do any logic change only refactoring

**Comment by RahulGautamSingh** [comment]
Test run: https://github.com/RahulGautamSingh-testing/r-m/pulls

**Review by secustor — APPROVED** [review]

**Pull request merged**